feat(query): Add a way to make bulk table queries

wmak opened the pull request:
- This adds a way so that endpoints can construct multiple TableQueries and fire them off in parallel instead of individually

wmak committed:
feat(query): Add a way to make bulk table queries

- This adds a way so that endpoints can construct multiple TableQueries
  and fire them off in parallel instead of individually

wmak commented on src/sentry/snuba/rpc_dataset_common.py:
```diff
@@ -163,33 +167,59 @@ def validate_granularity(
-    debug: bool = False,
-) -> EAPResponse:
+def run_bulk_table_queries(queries: dict[str, TableQuery]):
+    prepared_queries = {name: get_table_rpc_request(query) for name, query in queries.items()}
```
🤔 I sorta hate the tuple API in python lol, maybe i'll just make them a list of objects with name and query instead so its way more explicit

Zylphrex replied:
That works too, just something where the ordering is explicit rather than relying on people knowing that dictionaries are in insert order.

wmak committed:
ref: Make name a property of TableQuery so we're passing a list instead

- This is so we aren't relying on dictionary order which should make it
  clearer
